calico_cni_k8s_test.go

```diff
@@ -465,6 +466,7 @@ var _ = Describe("CalicoCni", func() {
 			          "nodename_file_optional": true,
 			 	  "ipam": {},
 					"kubernetes": {
+					  "enable_ip_addrs_no_ipam": true,
```

Comment:
I think my main bit of feedback is whether this belongs in the "kubernetes" section.

Imagining a world where we support this feature in other integrations, or where we want to enable / disable other features that aren't k8s specific.

So, we might want something more generic. WDYT about something like this?

```json
{
    "kubernetes": {},
    "feature_control": {
        "ip_addrs_no_ipam": false,
        "ip_addrs": true,
        "namespace_ipam": true,
    }
}
```